src/js/media-controller.js

```diff
@@ -496,8 +499,29 @@ const getMediaUIAttributesFrom = (child) => {
   );
 };
 
+const hasMediaUIProps = (mediaStateReceiverCandidate) => {
+  if (
+    mediaStateReceiverCandidate.nodeName?.includes('-')
+    && !!window.customElements.get(mediaStateReceiverCandidate.nodeName?.toLowerCase())
+    && !(mediaStateReceiverCandidate instanceof window.customElements.get(mediaStateReceiverCandidate.nodeName.toLowerCase()))
+  ) {
+    window.customElements.upgrade(mediaStateReceiverCandidate);
+  }
+  return MEDIA_UI_PROP_NAMES.some(propName => propName in mediaStateReceiverCandidate);
```

Comment:
Seeing this implemented, I'm feeling like it might start pushing on the safeness of the `media` prefix. No changes, just thinking about it.

Reply:
Yeah for sure, this came up in previous discussions. As long as we're diligent about it wrt attr/prop names in "built-ins" for HTML (an extension of https://www.w3.org/TR/design-principles/#attribute-reuse) and expect these conflicts to be the exception and not the rule for custom elements, we should be able to introduce some kind of "ignore" logic in our media state receiver detection/propagation in the future.